src/util.ts

```diff
@@ -58,28 +58,17 @@ export async function zipDir(
       core.info(`function source zipfile created: ${archive.pointer()} bytes`);
     });
     archive.pipe(output);
-    // Add dir to root of archive
-    getFiles(dirPath).forEach((filepath) => {
-      archive.glob(filepath, {
-        cwd: dirPath,
-        noglobstar: true,
-      });
-    });
+    // gcloudignore
+    const gIgnore = ignore().add(getGcloudIgnores(dirPath));
+    // Add files in dir to archive iff file not ignored
+    archive.directory(dirPath, false, (file) =>
```

Comment:
The performance is way better and the deployment works again. You could think about to only pass the callback to the `directory` function if the length of the returning array of `getGcloudIgnores(dirPath)` is greater than 0. This way you would save to check every file in cases where it's not even necessary. Because in my case I don't have a `gcloudignore` file but still every file is checked.

Reply:
Good point, added in https://github.com/google-github-actions/deploy-cloud-functions/pull/101/commits/03a901f3dbeb5a872d621ea352d5bf2ba23a459e